src/client/metadataApiDeploy.ts

```diff
@@ -428,7 +428,11 @@ export class MetadataApiDeploy extends MetadataTransfer<
     );
     // only do event hooks if source, (NOT a metadata format) deploy
     if (this.options.components) {
-      await lifecycle.emit('scopedPostDeploy', { deployResult, orgId: this.orgId } as ScopedPostDeploy);
+      // this may not be set if you resume a deploy so that `pre` is skipped.
+      if (!this.orgId) {
+        this.orgId = connection.getAuthInfoFields().orgId;
+      }
+      await lifecycle.emit('scopedPostDeploy', { deployResult, orgId: this.orgId } satisfies ScopedPostDeploy);
```

Comment:
![image](https://user-images.githubusercontent.com/76090802/222003660-edb165aa-553c-4d55-b622-88eefbc91423.png)
I had to lookup what satisfies does again.  Cool. new and shiny.  For my own understanding why do we need it in this situation? Doesn't the value passed already match the type? I'm not sure why there was an `as` in the first place.

Reply:
(unrelated to Gordon's comment) I don't like this entire section.  I don't know what "scopedPostDeploy" is supposed to mean and I think it's too narrow as an event fired from a library.  IMO the library should fire an event called "postDeploy" and the payload contains the data required for listeners to handle it or not.  Otherwise you run into situations where a client wants to subscribe to all deploy events from the library (not just source deploy, but also mdapi deploy) so you need another event for essentially the same thing.

Reply:
Steve: I generally agree...but...we had an existing postDeploy event. It was typed as `deployResult` (not an object containing a deployResult) so that it's not possible to add properties (like the orgId) without being a breaking change to publicly documented hooks events.

Scoped include the orgId, so that a listener (STL) not only listens for the events, but knows which org they belong to.

Reply:
Right, and I think the plugin event is not great either but it was already dealing with toolbelt's event history.  Since we're using core's Lifecycle singleton we need to use a good event naming pattern to avoid collisions, or not use it.  I think you could still name it `sdr:postDeploy`, always fire it and the payload is an object that has all the necessary data for handling.  Not part of this PR, and it would be a major version bump, but I think we should consider it especially if the need arises.